cps/tasks/metadata_extract.py

```diff
@@ -66,7 +67,10 @@ def _fetch_requested_urls(self, conn):
                      if "error" in self.columns
                      else "SELECT path, duration FROM media WHERE path LIKE 'http%'")
             rows = conn.execute(query).fetchall()
-            return {row[0]: {"duration": row[1]} for row in rows}
+            requested_urls = {row[0]: {"duration": row[1]} if row[1] is not None and row[1] > 0 else self.unavailable.append(row[0]) for row in rows}
+            for url in self.unavailable:
+                requested_urls.pop(url)
+            return requested_urls
```

Comment:
Totally untested code  that seems simpler and more readable than the list comprehension with conditionals.

```python
requested = {}
for path, duration in rows:
    if duration is not None and duration > 0:
      requested[path] = {"duration": duration};
   else:
      self.unavailable.append(path)
   return requested
```

Reply:
@deldesir I've in favor or simple + more readable if you can consider?!

Reply:
Done!  In 2 steps...

- PR #180
- https://github.com/iiab/calibre-web/commit/b0adaa0b26e984d3048a4183cf280dcb4a99d3e9